components/butterfly/src/rumor/service.rs

```diff
@@ -87,7 +87,7 @@ impl Service {
         package: &T,
         service_group: ServiceGroup,
         sys: SysInfo,
-        cfg: Option<&toml::value::Table>,
+        cfg: Option<Vec<u8>>,
```

Comment:
We might be able to simplify things by using`toml::Value` instead. That's what's recommended in https://github.com/alexcrichton/toml-rs/issues/142, and clicking through to the code in question (current code is [here](https://github.com/alexcrichton/toml-rs/blob/56f9afb4d2abdf38ed3caf9f819ac8bacc290940/src/value.rs#L421-L451) actually shows that we have copied that code into our module (presumably from before that code was actually released).

We might be able to simplify the representation, fix the bug, _and_ ditch some complex code that we can now get for free.

Reply:
Now that we're dealing with the weird TOML table ordering issue, is there any reason we expect this should fail? I feel like this existing bug would've been a lot harder to find and fix it we handled the error before instead of panicking.

In other words, I think panicking here is appropriate unless we expect there to be an error case we want to handle more gracefully.

Reply:
I don't think it should now fail given we're using the right way from the crate instead of doing it in our code.

Reply:
So, do we agree that panicking is appropriate, then?

Reply:
Yep.